File: packages/ts/test/client.test.ts
import { describe, expect, it } from 'bun:test';
import { HubflyClient } from '../src/index.js';

describe('HubflyClient', () => {
  it('should initialize with custom options and expose all modules', () => {
    const client = new HubflyClient({
      token: 'test-token',
      baseUrl: 'http://localhost:8787',
    });

    expect(client.baseUrl).toBe('http://localhost:8787');
    expect(client.auth).toBeDefined();
    expect(client.projects).toBeDefined();
    expect(client.projects.containers).toBeDefined();
    expect(client.projects.loadBalancers).toBeDefined();
    expect(client.projects.network).toBeDefined();
    expect(client.projects.ports).toBeDefined();
    expect(client.projects.registry).toBeDefined();
    expect(client.projects.subdomains).toBeDefined();
    expect(client.projects.team).toBeDefined();
    expect(client.projects.tunnels).toBeDefined();
    expect(client.projects.volumes).toBeDefined();
    expect(client.projects.webhooks).toBeDefined();
    expect(client.projects.hibernation).toBeDefined();
    expect(client.organizations).toBeDefined();
    expect(client.domains).toBeDefined();
    expect(client.gpu).toBeDefined();
    expect(client.billing).toBeDefined();
    expect(client.system).toBeDefined();
    expect(client.regions).toBeDefined();
    expect(client.templates).toBeDefined();
  });
});

File: packages/go/client_test.go
package hubfly_test

import (
	"testing"

	"github.com/hubfly/hubfly-sdk/go"
)

func TestNewClient(t *testing.T) {
	client := hubfly.NewClient(
		hubfly.WithToken("test-token"),
		hubfly.WithBaseURL("http://localhost:8787"),
	)

	if client == nil {
		t.Fatal("expected non-nil client")
	}

	if client.Auth == nil {
		t.Error("expected Auth service to be initialized")
	}
	if client.Projects == nil {
		t.Error("expected Projects service to be initialized")
	}
	if client.Containers == nil {
		t.Error("expected Containers service to be initialized")
	}
	if client.Regions == nil {
		t.Error("expected Regions service to be initialized")
	}
	if client.Templates == nil {
		t.Error("expected Templates service to be initialized")
	}
	if client.LoadBalancers == nil {
		t.Error("expected LoadBalancers service to be initialized")
	}
	if client.Network == nil {
		t.Error("expected Network service to be initialized")
	}
	if client.Ports == nil {
		t.Error("expected Ports service to be initialized")
	}
	if client.Registry == nil {
		t.Error("expected Registry service to be initialized")
	}
	if client.Subdomains == nil {
		t.Error("expected Subdomains service to be initialized")
	}
	if client.Team == nil {
		t.Error("expected Team service to be initialized")
	}
	if client.Tunnels == nil {
		t.Error("expected Tunnels service to be initialized")
	}
	if client.Volumes == nil {
		t.Error("expected Volumes service to be initialized")
	}
	if client.Webhooks == nil {
		t.Error("expected Webhooks service to be initialized")
	}
	if client.Hibernation == nil {
		t.Error("expected Hibernation service to be initialized")
	}
}
